tests/python/relax/test_codegen_coreml.py

```diff
@@ -0,0 +1,315 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+import numpy as np
+import pytest
+
+import tvm
+import tvm.testing
+from tvm import relax
+from tvm.relax.backend.contrib.coreml import partition_for_coreml
+
+requires_coremltools = tvm.testing.requires_package("coremltools")
+target, dev = "llvm", tvm.cpu()
+
+
+def _has_xcode():
+    try:
+        tvm.contrib.xcode.xcrun([])
+        return True
+    except FileNotFoundError:
+        pass
+
+    return False
+
+
+@pytest.mark.skipif(not _has_xcode(), reason="Xcode is not available")
+@tvm.testing.uses_gpu
+@requires_coremltools
+def verify(mod, inputs):
+    mod1 = partition_for_coreml(mod)
+    mod1 = relax.transform.RunCodegen()(mod1)
+    assert relax.analysis.well_formed(mod1)
+    assert mod1.attrs, "Should exist if offloaded successfully."
+    assert "external_mods" in mod1.attrs, "Should exist if offloaded successfully."
+    mod1 = relax.transform.LegalizeOps()(mod1)
+    assert relax.analysis.well_formed(mod1)
+
+    ex1 = relax.build(mod1, target=target)
+    vm1 = relax.VirtualMachine(ex1, dev, profile=True)
+    out1 = vm1["main"](*inputs)
+
+    mod2 = relax.transform.LegalizeOps()(mod)
+    ex2 = relax.build(mod2, target=target)
+    vm2 = relax.VirtualMachine(ex2, dev, profile=True)
+    out2 = vm2["main"](*inputs)
+
+    tvm.testing.assert_allclose(out1.numpy(), out2.numpy(), rtol=1e-3, atol=1e-3)
```

Comment:
Please note that current error margin is bit higher than our usual setting for CPU computations (`rtol=1e-7, atol=1e-7`), but lower than the one for GPU libraries (cutlass: `rtol=1e-2, atol=1e-2`)